lua/autorun/server/sv_daily_restart.lua

```diff
@@ -1,67 +1,88 @@
-    timer.Create("CFC_DailyRestart", 1, 0, function()
-        -- TODO: Check to make sure that the latest restart time wasn't recently
-        -- if countingDownToRestart then return
-        -- if waitingForServerToEmpty then return
+    if isLastScheduledRestartValid then
+        if os.difftime(os.time(), lastScheduledRestart) <= scheduledTimeThreshold then return end
+    end
+
+    local formattedTime = os.date("*t", currentTime)
 
-        theTime = os.date("*t", os.time())
+    -- TODO: Also check each second within 1 hour leading up to the restart time. If the server is empty during this time, issue the restart and ensure we don't try to restart again at the desired time
+    --
+    -- TODO (Future): Perhaps if there are fewer than N players on the server leading up to the restart time, allow them to vote to restart now and get it out of the way
 
-        -- TODO: Also check each second within 1 hour leading up to the restart time. If the server is empty during this time, issue the restart and ensure we don't try to restart again at the desired time
-        --
-        -- TODO (Future): Perhaps if there are fewer than N players on the server leading up to the restart time, allow them to vote to restart now and get it out of the way
+    if not countingDownToRestart then
+        local isHourEquivalentToDesired = (formattedTime.hour == desiredRestartTime)
+        local secondThreshold = 30
+        local isSecondWithinThreshold = (formattedTime.sec <= secondThreshold)
 
-        if (theTime.hour == desiredRestartTime) and (theTime.sec <= 30) then
+        if isHourEquivalentToDesired and isSecondWithinThreshold then
             local plyCount = table.Count(player.GetHumans())
 
             if plyCount == 0 then
-                timer.Remove("CFC_DailyRestart")
-                restartServer()
+                CFCDailyRestart:restartServer()
             end
 
-            timer.Remove("CFC_DailyRestart")
             -- When the server is not empty..
             -- TODO: Wait up to 45 minutes for players to leave organically
             -- If server still isn't empty by then, countdown restart
             -- You'll probably want to make a new function called waitForServerToEmpty() or something here and make sure the timer isn't triggering in the meantime
             -- You may want to keep a variable called "waitingForServerToEmpty" for this function to look at and no-op if so
             -- When the server is still not empty, you'll need to start the shutdown process. "countingDownToRestart" variable will tell this function not to re-initiate the process
-            --countingDownToRestart = 1
-            alertClientsOfServerRestart()
+            CFCDailyRestart:alertClientsOfServerRestart()
+            CFCDailyRestart.countingDownToRestart = true
         end
-    end)
-end
+
+        return
+    end
+
+    if os.difftime(currentTime, CFCDailyRestart.restartAt) % 60 == 0 then -- Quick fix for alerting players of restart
```

Comment:
It's probably acceptable to break everything in this nest out into its own function.